.editorconfig

```diff
@@ -156,7 +161,479 @@ csharp_space_between_square_brackets = false
+dotnet_diagnostic.CA5394.severity = warning
+# Remove unnecessary using directives
+dotnet_diagnostic.IDE0005.severity = warning
+# Fix formating
+dotnet_diagnostic.IDE0055.severity = warning
+# FileHeaderMismatch
+dotnet_diagnostic.IDE0073.severity = warning
+
+# Rules for TemplateEngine
+[src/Cli/**{Microsoft.TemplateEngine.*,dotnet-new*}/**.cs]
+# Use Length/Count property instead of Count() when available
+dotnet_diagnostic.CA1829.severity = warning
+# Consider calling ConfigureAwait on the awaited task
+dotnet_diagnostic.CA2007.severity = warning
+# Forward the 'CancellationToken' parameter to methods that take one
+dotnet_diagnostic.CA2016.severity = warning
+
+# Enable StyleCop analyzer in src/Cli/dotnet for TempleteEngine but not affect others
+[src/Tests/Common/Program.cs]
+# SA1205: Partial elements should declare access
+dotnet_diagnostic.SA1205.severity = none
+# SA1516: Elements should be separated by blank line
+dotnet_diagnostic.SA1516.severity = none
+[{src/Common/**.cs,src/Cli/dotnet/**.cs,src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/FXVersion.cs}]
+# MakeLocalFunctionStatic
+dotnet_diagnostic.IDE0062.severity = none
+# Remove redundant nullable directive
+dotnet_diagnostic.IDE0240.severity = none
+# Disable StyleCop Analyzer for src/Cli/dotnet
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.OrderingRules.severity = none
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.LayoutRules.severity = none
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.SpacingRules.severity = none
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.ReadabilityRules.severity = none
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.NamingRules.severity = none
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.MaintainabilityRules.severity = none
+# Refer to tuple elements by name
+dotnet_diagnostic.SA1142.severity = none
+# Tuple element names should use correct casing
+dotnet_diagnostic.SA1316.severity = none
+
+# Enable StyleCop analyzer in src/Cli/dotnet/commands/dotnet-new for TempleteEngine
+[src/Cli/dotnet/commands/dotnet-new/**.cs]
+# MakeLocalFunctionStatic
+dotnet_diagnostic.IDE0062.severity = warning
+# Remove redundant nullable directive
+dotnet_diagnostic.IDE0240.severity = warning
+# Disable StyleCop Analyzer for src/Cli/dotnet
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.OrderingRules.severity = warning
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.LayoutRules.severity = warning
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.SpacingRules.severity = warning
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.ReadabilityRules.severity = warning
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.NamingRules.severity = warning
+dotnet_analyzer_diagnostic.category-StyleCop.CSharp.MaintainabilityRules.severity = warning
+# Refer to tuple elements by name
+dotnet_diagnostic.SA1142.severity = warning
+# Tuple element names should use correct casing
+dotnet_diagnostic.SA1316.severity = warning
```

Comment:
imo, enabling StyleCop for these files is not required. Let's use sdk set of rules in `dotnet`